README.md

```diff
@@ -78,7 +78,9 @@ Add the following to your `.bashrc` (or `.profile` on Mac):
 
 ```
 function _update_ps1() {
-    PS1="$(~/go/bin/powerline-go -error $?)"
+    if [ -e $GOPATH/bin/powerline-go ]; then
```

Comment:
As of `go help gopath` the `GOPATH` environment variable supports a list of places to look for Go code.
If this is the case, this code will fail. You will need some additional `bash` magic to split `$GOPATH` by the seperator `: `.

Reply:
Even worse, Go might also look in $GOBIN.

We should make a general function to find the path that is wanted, and add that to the install scripts

Reply:
From my point of view, there may be an ideal solution, but having something which avoids the situation of the binary being missing and the prompt being gone seems more important in the first instance :)

Reply:
SInce the conditional `-e` only checks for the existence of the file, would it be useful to change that to `-x` to check for the existence and execute bit set for the file?

Reply:
Actually, it might be simpler to just try to run powerline-go, and check if that exits with code 0. If not, show the fallback.

Reply:
Something like
     
    function _update_ps1() {
        $GOPATH/bin/powerline-go
        if [[ "${?}" -eq 0 ]]; then
            PS1="$($GOPATH/bin/powerline-go -error $?)"
        fi
    }

Reply:
Even simpler:

```bash
function _update_ps1() {
    $_TMP_PS1=$($GOPATH/bin/powerline-go -error $?)
    if [[ "${?}" -eq 0 ]]; then
        PS1=$_TMP_PS1
    fi
}
```

Reply:
The first line of the function: `_TMP_PS1= $` since I don't think you have that variable declared earlier, bash will try to interpret that as an existing variable and not a new declaration of a new variable within the function.

Reply:
Sure — I'm writing this at 3am on mobile, so I don't exactly expect it to be perfect — I'm mostly just outlining the general idea of where we should try to go.

Reply:
Yeah, and I'm on a non-linux station at the moment that I can't test, but that looks good, puts the test in a subshell so there's no worry about it existing after the function exits. Good thought!